torch/csrc/jit/tensorexpr/tensorexpr_tutorical.py

```diff
@@ -0,0 +1,219 @@
+# transform the stmt to 1) expand reduce operations, and 2) flatten buffer
+# dimemsions. In this example, there are no reduces or multi-dimentional
+# buffers, thus the stmt remaining the same.
+loopnest = te.LoopNest([C])
+loopnest.prepare_for_codegen()
+
+# Print out the transformed stmt.
+print("The stmt before codegen:")
+print(loopnest.root_stmt())
+
+# The stmt before codegen:
+# {
+#   for (int v = 0; v < 32; v++) {
+#     C[v] = (A[v]) + (B[v]);
+#   }
+# }
+
+# Optional: simplify the stmt.
+# TODO: Should we add 'simplify' to 'prepare_for_codegen'? This seems to be a
+# common practise we do before codegen.
+stmt = te.simplify(loopnest.root_stmt())
+
+# Now we can get to codegen. Here, we use the IR Evaluator as the backend to
+# lower the stmt to assembly code. Besides the stmt, we also need to specify
+# the inputs and output data for codegen.
+cg = te.construct_codegen("ir_eval", stmt, [A, B, C])
+
+# We have generated native code in TE to add two tensors. Let's try it out!
+# Prepare the input/output tensors, and call TE kernel. The result is saved in
+# the output tensor `tC`.
+tA = torch.randn(32, dtype=torch.float32)
+tB = torch.randn(32, dtype=torch.float32)
+tC = torch.empty(32, dtype=torch.float32)
+cg.call_raw([tA.data_ptr(), tB.data_ptr(), tC.data_ptr()])
+
+# We can check the correctness of the results as follows.
+torch.testing.assert_close(tA + tB, tC)
+
+# Hoory, now you have learned how to create and run TE kernel!
+
+
+# 2. Construct your own TE IR
+
+# TE constants
+c = te.ExprHandle.int(5)
+f = te.ExprHandle.float(5.2)
+print("\nTE constants:")
+print(c, f)
+
+# TE integer variables
+m = te.VarHandle("m", torch.int32)
+n = te.VarHandle("n", torch.int32)
+print("\nTE integer variables:")
+print(m, n)
+
+# TE arithmeic expression
+# no needs to convert 5 to a TE constant before the addition; it will
+# be converted to the same type of m when performing the addition operation
+mc = m + 5
+mn = m + n
+e = te.sin(mc * mn)
+print("\nTE arithmetic expression:")
+print(e)
+
+# TE buffer variables
+A = te.BufHandle("A", [te.ExprHandle.int(32), te.ExprHandle.int(32)], torch.float32)
+B = te.BufHandle("B", [te.ExprHandle.int(32), te.ExprHandle.int(32)], torch.float32)
+print("\nTE buffer variables:")
+print(A, B)
+
+# TE load expression
+i = te.VarHandle("i", torch.int32)
+j = te.VarHandle("j", torch.int32)
+print("\nTE load expression:")
+print(A.load([i, j]))
+
+# TE store statement
+C = te.BufHandle("C", [te.ExprHandle.int(32), te.ExprHandle.int(32)], torch.float32)
+store = C.store([i, j], A.load([i, j]) + B.load([i, j]))
+print("\nTE store statement:")
+print(store)
+
+# TE block statement
+block = te.Block([store])
+print("\nTE block statement:")
+print(block)
+
+# TE conditional statement
+cond = te.Cond.make(i > j, block, None)
+print("\nTE cond statement:")
+print(cond)
+
+# TE for statement
+forst = te.For.make(j, te.ExprHandle.int(0), te.ExprHandle.int(32), block)
```

Comment:
Did you plan to use `cond` instead of `block` here? Currently `block` is a child of both `cond` and `forst` IIUC.